src/components/ActivityForm.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+
+export default class ActivityForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classNo: props.activity ? props.activity.classNo : '',
+      day: props.activity ? props.activity.day : '',
+      name: props.activity ? props.activity.name : '',
+      room: props.activity ? props.activity.room : '',
+      teacher: props.activity ? props.activity.teacher : '',
+      error: ''
+    };
+  };
+  onNameChange = (e) => {
+    const name = e.target.value;
+    this.setState(() => ({name}));
+  };
+  onTeacherChange = (e) => {
+    const teacher = e.target.value;
+    this.setState(() => ({teacher}));
```

Comment:
if you are not using `prevState` you can pass here the object. Like:
`onTeacherChange = e => ({ teachher: e.target.value })`